apps/frontend/src/store/report_intake.ts

```diff
@@ -93,14 +132,120 @@ export class InspecIntake extends VuexModule {
+      return parsed.platform !== undefined;
+    } catch {
+      // HDF isn't valid json, we have a different format
+      return false;
+    }
+  }
+
+  @Action
+  async convertToHdf(convertOptions: {
+    file: FileLoadOptions;
+    data: string;
+  }): Promise<ExecJSON.Execution | ExecJSON.Execution[] | void> {
+    try {
+      const parsed = JSON.parse(convertOptions.data);
+      // If the data passed is valid json, try to match up known keys
+      const typeGuess = await this.guessType(parsed);
+      if (typeGuess === 'jfrog') {
+        return new JfrogXrayMapper(convertOptions.data).toHdf();
+      } else if (typeGuess === 'zap') {
+        return new ZapMapper(convertOptions.data).toHdf();
+      } else if (typeGuess === 'nikto') {
+        return new NiktoMapper(convertOptions.data).toHdf();
+      } else if (typeGuess === 'sarif') {
+        return new SarifMapper(convertOptions.data).toHdf();
+      } else if (typeGuess === 'snyk') {
+        return new SnykResults(convertOptions.data).toHdf();
+      }
+    } catch {
+      // If we don't have JSON, fall back to checking strings inside of the text
+      // Nessus
+      if (convertOptions.file.file.name.toLowerCase().endsWith('.nessus')) {
+        return new NessusResults(convertOptions.data).toHdf();
+      }
+      // XCCDF Results
+      else if (
+        // If the file contains the Checklist schema or has xccdf in the filename
+        convertOptions.data.indexOf(
+          'schemaLocation="http://checklists.nist.gov/xccdf'
+        ) !== -1 ||
+        convertOptions.file.file.name.toLowerCase().indexOf('xccdf') !== -1
+      ) {
+        return new XCCDFResultsMapper(convertOptions.data).toHdf();
+      }
+      // Burp Suite
+      else if (convertOptions.data.indexOf('issues burpVersion') !== -1) {
+        return new BurpSuiteMapper(convertOptions.data).toHdf();
+      }
+      // Scoutsuite
+      else if (convertOptions.data.indexOf('scoutsuite_results') !== -1) {
+        return new ScoutsuiteMapper(convertOptions.data).toHdf();
+      }
+      // DBProtect
+      else if (
+        convertOptions.data.indexOf('Policy') !== -1 &&
+        convertOptions.data.indexOf('Job Name') !== -1 &&
+        convertOptions.data.indexOf('Check ID') !== -1 &&
+        convertOptions.data.indexOf('Result Status')
+      ) {
+        return new DBProtectMapper(convertOptions.data).toHdf();
+      } else if (convertOptions.data.indexOf('netsparker-enterprise')) {
+        return new NetsparkerMapper(convertOptions.data).toHdf();
+      }
+    }
+    return SnackbarModule.failure(
+      'Invalid file uploaded, no fingerprints matched.'
+    );
+  }
+
+  @Action
+  async guessType(parsedJSON: Record<string, unknown>): Promise<string> {
+    // Find the fingerprints that have the most matches
+    const fingerprinted = Object.entries(fileTypeFingerprints).reduce(
+      (a, b) => {
+        return a[1].filter((value) => _.get(parsedJSON, value)).length >
```

Comment:
This logic is a bit fussy for me to realize how the reduction is occurring. I believe there is a risk here in returning the wrong guess.

Reply:
This works by counting the number of matches for every fingerprint in `fileTypeFingerprints`. The type that has the most matches is determined to be the filetype.

I don't believe that this has much of a chance of choosing the wrong value since the only fingerprints that overlap are the 'vulnerabilities' field in Nikto and Snyk, both of which have other fingerprints to avoid such a scenario.